cli/lib/appdata.ts

```diff
@@ -122,14 +131,24 @@ export async function getAuthToken(): Promise< NonNullable< UserData[ 'authToken
 
 export async function getSiteByFolder(
 	siteFolder: string
-): Promise< z.infer< typeof siteSchema > > {
+): Promise< z.infer< typeof siteSchema > | null > {
 	const userData = await readAppdata();
 	const sites = userData.sites ?? [];
 	const site = sites.find( ( site ) => site.path === siteFolder );
 
 	if ( ! site ) {
-		throw new LoggerError( __( 'The specified folder is not added to Studio.' ) );
+		return null;
 	}
 
 	return site;
 }
+
+export function getNewSiteAppData( siteFolder: string ): z.infer< typeof siteSchema > {
```

Comment:
```suggestion
export function getNewSitePartial( siteFolder: string ) {
```
I feel like we could clarify the name here. I would also let TS automatically intuit the return type.